File: 01-palindrome-number/plaindrome-number.c
#include <stdio.h>

int reverse_number(int num) {
  int remainder, reverse = 0;

  while(num != 0) {
    remainder = num % 10; 
    reverse = reverse * 10 + remainder;
    num /= 10;
  }

  return reverse;
}

int main() {
  
  int input = 0;

  printf("Enter a number: ");
  scanf("%d",&input);

  int reverse = reverse_number(input);

  if (input == reverse) {
    printf("This is a plaindrome number\n");
  }
  else {
    printf("This is not a plaindrome number\n");
  }
  

  return 0;
}

File: 01-palindrome-number/plaindrome-number.rs
use std::io;

fn main() {

    let mut input_text = String::new();

    println!("Input a number:");
    io::stdin().read_line(&mut input_text).expect("Failed to read from stdin");

    let num = input_text.trim().parse::<i32>().expect("Invalid input");

    let reveresed = reverse_num(num);

    if num < 0 {
        println!("This is not a plaindrome number!!!");
    }
    else if num == reveresed {
        println!("This is a plaindrome number!!");
    }
    else {
        println!("This is not a plaindrome number!!!");
    }

}

fn reverse_num(mut num: i32) -> i32 {
    let mut remainder: i32;
    let mut reverse: i32 = 0;

    while num != 0 {
        remainder = num % 10; 
        reverse = reverse * 10 + remainder;
        num /= 10;
    }

    return reverse;
    
}

